c2rust-refactor/src/analysis/ownership/mod.rs

```diff
@@ -196,6 +197,46 @@ fn analyze_inter<'lty, 'tcx>(cx: &mut Ctxt<'lty, 'tcx>) {
     inter_cx.finish();
 }
 
+fn is_mut_t(ty: &TyS) -> bool {
+    if let TyKind::RawPtr(mut_ty) = ty.sty {
+        if mut_ty.mutbl == Mutability::MutMutable {
+            if let TyKind::Param(param_ty) = mut_ty.ty.sty {
+                return param_ty.name.as_str().get() == "T";
+            }
+        }
+    }
+
+    false
+}
+
+fn register_std_constraints<'a, 'tcx, 'lty>(
+    ctxt: &mut Ctxt<'lty, 'tcx>,
+    tctxt: TyCtxt<'tcx>,
+) {
+    for (def_id, func_summ) in ctxt.funcs_mut() {
+        // #[ownership_constraints(le(WRITE, _0), le(WRITE, _1), le(_0, _1))]
+        // fn offset<T>(self: *mut T, _: isize) -> *mut T;
+        if func_summ.sig.inputs.len() == 2 {
+            let param0_is_mut_t = is_mut_t(func_summ.sig.inputs[0].ty);
+            let param1_is_isize = if let TyKind::Int(int_ty) = func_summ.sig.inputs[1].ty.sty {
+                int_ty == IntTy::Isize
+            } else {
+                false
+            };
+            let ret_is_mut_t = is_mut_t(func_summ.sig.output.ty);
+            if param0_is_mut_t && param1_is_isize && ret_is_mut_t {
+                if tctxt.def_path(*def_id).to_string_no_crate() == "::ptr[0]::{{impl}}[1]::offset[0]" {
```

Comment:
Why don't you check this condition early? It's the most restrictive of the nested conditions.

Reply:
Gooood point. I think it was the last check I added, so checking early slipped my mind